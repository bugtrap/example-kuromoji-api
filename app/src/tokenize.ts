import { KuromojiToken, getTokenizer } from "kuromojin";

export async function tokenize(text: string): Promise<KuromojiToken[]> {
    const tokenizer = await getTokenizer();
    return tokenizer.tokenize(text);
}
