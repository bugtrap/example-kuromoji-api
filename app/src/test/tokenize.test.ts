import { tokenize } from "../tokenize";

describe("解析", () => {
    it("簡単な動作テスト", async () => {
        const text = "テストです。";
        const words = await tokenize(text);

        expect(words).toMatchSnapshot();
    });
});
