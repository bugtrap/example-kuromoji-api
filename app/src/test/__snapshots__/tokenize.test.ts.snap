// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`解析 簡単な動作テスト 1`] = `
[
  {
    "basic_form": "テスト",
    "conjugated_form": "*",
    "conjugated_type": "*",
    "pos": "名詞",
    "pos_detail_1": "サ変接続",
    "pos_detail_2": "*",
    "pos_detail_3": "*",
    "pronunciation": "テスト",
    "reading": "テスト",
    "surface_form": "テスト",
    "word_id": 698600,
    "word_position": 1,
    "word_type": "KNOWN",
  },
  {
    "basic_form": "です",
    "conjugated_form": "基本形",
    "conjugated_type": "特殊・デス",
    "pos": "助動詞",
    "pos_detail_1": "*",
    "pos_detail_2": "*",
    "pos_detail_3": "*",
    "pronunciation": "デス",
    "reading": "デス",
    "surface_form": "です",
    "word_id": 23760,
    "word_position": 4,
    "word_type": "KNOWN",
  },
  {
    "basic_form": "。",
    "conjugated_form": "*",
    "conjugated_type": "*",
    "pos": "記号",
    "pos_detail_1": "句点",
    "pos_detail_2": "*",
    "pos_detail_3": "*",
    "pronunciation": "。",
    "reading": "。",
    "surface_form": "。",
    "word_id": 90940,
    "word_position": 6,
    "word_type": "KNOWN",
  },
]
`;
